src/Rings/FreeAssAlgIdeal.jl

```diff
@@ -98,24 +110,26 @@ true
 ```
 """
 function ideal_membership(a::FreeAssAlgElem, I::FreeAssAlgIdeal, deg_bound::Int=-1)
-  return ideal_membership(a, IdealGens(gens(I)), deg_bound)
+  isdefined(I, :gb) && (I.deg_bound == -1 || I.deg_bound >= deg_bound) && return iszero(normal_form(a, I.gb))
+  groebner_basis(I, deg_bound)
+  return ideal_membership(a, I, deg_bound)
 end
 function ideal_membership(a::FreeAssAlgElem, I::IdealGens{<:FreeAssAlgElem}, deg_bound::Int=-1)
   return ideal_membership(a, collect(I), deg_bound)
 end
 function ideal_membership(a::FreeAssAlgElem, I::Vector{<:FreeAssAlgElem}, deg_bound::Int=-1)
   R = parent(a)
   @assert all(x -> parent(x) == R, I) "parent mismatch"
-  gb = groebner_basis(I, deg_bound; protocol=false)
-  deg_bound = max(maximum(total_degree.(gb)),total_degree(a))
-  lpring, _ = _to_lpring(R, deg_bound)
+  gb = groebner_basis(I, deg_bound)
+  return iszero(normal_form(a, gb))
+end
 
-  lp_I = Singular.Ideal(lpring, lpring.(gb))
-  return iszero(reduce(lpring(a), lp_I))
+function Base.in(a::FreeAssAlgElem, I::FreeAssAlgIdeal)
+  return ideal_membership(a, I)
 end
 
-function Base.in(a::FreeAssAlgElem, I::FreeAssAlgIdeal, deg_bound::Int)
-  return ideal_membership(a, I, deg_bound)
+function Base.in(I1, I2::FreeAssAlgIdeal)
```

Comment:
Maybe this should rather be `is_subset` instead of `in`?

Reply:
I think this should be as similar as possible (w.r.t naming an interface) to ideals in MPolyRing. (not sure right now how they do this particular thing though)